pkg/components/power-monitor/deployment.go

```diff
@@ -450,6 +661,64 @@ func newPowerMonitorContainer(pmi *v1alpha1.PowerMonitorInternal) corev1.Contain
 	}
 }
 
+func newKubeRBACProxyContainer(pmi *v1alpha1.PowerMonitorInternal) corev1.Container {
+	deployment := pmi.Spec.Kepler.Deployment
+	return corev1.Container{
+		Name:  KubeRBACProxyContainerName,
+		Image: deployment.KubeRbacProxyImage,
+		Args: []string{
+			fmt.Sprintf("--secure-listen-address=0.0.0.0:%d", SecurePort),
+			fmt.Sprintf("--upstream=http://127.0.0.1:%d", PowerMonitorDSPort),
+			fmt.Sprintf("--auth-token-audiences=%s.%s.svc", pmi.Name, pmi.Namespace()),
+			fmt.Sprintf("--config-file=%s/config.yaml", KubeRBACProxyConfigMountPath),
+			fmt.Sprintf("--tls-cert-file=%s/tls.crt", PowerMonitorTLSMountPath),
+			fmt.Sprintf("--tls-private-key-file=%s/tls.key", PowerMonitorTLSMountPath),
+			"--allow-paths=/metrics",
+			"--logtostderr=true",
+			"--v=10",
+		},
+		Ports: []corev1.ContainerPort{{
+			Name:          SecurePortName,
+			ContainerPort: int32(SecurePort),
+		}},
+		Resources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("1m"),
+				corev1.ResourceMemory: resource.MustParse("15Mi"),
+			},
+		},
+		SecurityContext: &corev1.SecurityContext{
+			AllowPrivilegeEscalation: ptr.To(false),
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{"ALL"},
+			},
+		},
+		TerminationMessagePolicy: corev1.TerminationMessageFallbackToLogsOnError,
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: SecretKubeRBACProxyConfigName, MountPath: KubeRBACProxyConfigMountPath, ReadOnly: true},
+			{Name: SecretTLSCertName, MountPath: PowerMonitorTLSMountPath, ReadOnly: true},
+		},
+	}
+}
+
+func createKubeRBACConfig(serviceAccountNames []string) string {
+	var config k8s.KubeRBACProxyConfig
+
+	for _, serviceAccountName := range serviceAccountNames {
+		newRule := k8s.StaticRule{
+			Path:            "/metrics",
+			ResourceRequest: false,
+			User: k8s.User{
+				Name: fmt.Sprintf("system:serviceaccount:%s", serviceAccountName),
+			},
+			Verb: "get",
+		}
+		config.Authorization.StaticRules = append(config.Authorization.StaticRules, newRule)
+	}
+	yamlData, _ := yaml.Marshal(&config)
```

Comment:
lets not ignore (any) error .. can we propagate it or log it some how if it isn't easy, lets file an issue and the address this in a generic way.